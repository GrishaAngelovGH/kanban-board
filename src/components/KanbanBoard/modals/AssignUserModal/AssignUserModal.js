import { useState } from "react"

import Modal from "components/Modal"
import UserList from "components/KanbanBoard/UserList"

const AssignUserModal = ({ show, task, onClose, onConfirm }) => {
  const [ids, setIds] = useState([])

  return (
    <Modal
      show={show}
      onClose={onClose}
      onConfirm={onConfirm}
      title={<span>Assign users to the <span className="fw-bold">"{task?.title}"</span> task</span>}
      disabledConfirm={!ids.length}
      body={
        <UserList onUpdate={setIds} />
      }
    />
  )
}

export default AssignUserModal