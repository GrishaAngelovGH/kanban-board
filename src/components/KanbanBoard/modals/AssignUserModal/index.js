import AssignUserModal from "./AssignUserModal"

export default AssignUserModal