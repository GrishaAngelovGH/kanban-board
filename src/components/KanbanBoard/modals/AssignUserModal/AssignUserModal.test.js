import { render } from "@testing-library/react"

import AssignUserModal from "./AssignUserModal"

test("should render AssignUserModal component", () => {
  const view = render(<AssignUserModal show task={{ id: 1, title: "Title", description: "Description" }} />)

  expect(view).toMatchSnapshot()
})