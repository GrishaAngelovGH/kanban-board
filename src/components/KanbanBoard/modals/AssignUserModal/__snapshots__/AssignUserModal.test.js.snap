// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AssignUserModal component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="modal-open"
    data-rr-ui-modal-open=""
    style="padding-right: 1024px;"
  >
    <div />
    <div
      class="fade modal-backdrop show"
    />
    <div
      aria-modal="true"
      class="fade modal show"
      role="dialog"
      style="display: block; padding-right: 0px;"
      tabindex="-1"
    >
      <div
        class="modal-dialog"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <div
              class="modal-title h4"
            >
              <span>
                Assign users to the 
                <span
                  class="fw-bold"
                >
                  "
                  Title
                  "
                </span>
                 task
              </span>
            </div>
          </div>
          <div
            class="modal-body"
          >
            <div
              class="list-group"
            >
              <button
                class="list-group-item list-group-item-action"
              >
                <div
                  class="row align-items-center"
                >
                  <div
                    class="col-2"
                  >
                    <img
                      alt="assigned-user"
                      class="rounded-circle"
                      src="alexander-reed.jpg"
                      width="50"
                    />
                  </div>
                  <div
                    class="col-8"
                  >
                    <p
                      class="m-0 fw-bold"
                    >
                      Alexander Reed
                    </p>
                  </div>
                  <div
                    class="col-2"
                  />
                </div>
              </button>
              <button
                class="list-group-item list-group-item-action"
              >
                <div
                  class="row align-items-center"
                >
                  <div
                    class="col-2"
                  >
                    <img
                      alt="assigned-user"
                      class="rounded-circle"
                      src="ava-roberts.jpg"
                      width="50"
                    />
                  </div>
                  <div
                    class="col-8"
                  >
                    <p
                      class="m-0 fw-bold"
                    >
                      Ava Roberts
                    </p>
                  </div>
                  <div
                    class="col-2"
                  />
                </div>
              </button>
              <button
                class="list-group-item list-group-item-action"
              >
                <div
                  class="row align-items-center"
                >
                  <div
                    class="col-2"
                  >
                    <img
                      alt="assigned-user"
                      class="rounded-circle"
                      src="ethan-thompson.jpg"
                      width="50"
                    />
                  </div>
                  <div
                    class="col-8"
                  >
                    <p
                      class="m-0 fw-bold"
                    >
                      Ethan Thompson
                    </p>
                  </div>
                  <div
                    class="col-2"
                  />
                </div>
              </button>
              <button
                class="list-group-item list-group-item-action"
              >
                <div
                  class="row align-items-center"
                >
                  <div
                    class="col-2"
                  >
                    <img
                      alt="assigned-user"
                      class="rounded-circle"
                      src="isabella-foster.jpg"
                      width="50"
                    />
                  </div>
                  <div
                    class="col-8"
                  >
                    <p
                      class="m-0 fw-bold"
                    >
                      Isabella Foster
                    </p>
                  </div>
                  <div
                    class="col-2"
                  />
                </div>
              </button>
              <button
                class="list-group-item list-group-item-action"
              >
                <div
                  class="row align-items-center"
                >
                  <div
                    class="col-2"
                  >
                    <img
                      alt="assigned-user"
                      class="rounded-circle"
                      src="noah-anderson.jpg"
                      width="50"
                    />
                  </div>
                  <div
                    class="col-8"
                  >
                    <p
                      class="m-0 fw-bold"
                    >
                      Noah Anderson
                    </p>
                  </div>
                  <div
                    class="col-2"
                  />
                </div>
              </button>
              <button
                class="list-group-item list-group-item-action"
              >
                <div
                  class="row align-items-center"
                >
                  <div
                    class="col-2"
                  >
                    <img
                      alt="assigned-user"
                      class="rounded-circle"
                      src="sophia-campbell.jpg"
                      width="50"
                    />
                  </div>
                  <div
                    class="col-8"
                  >
                    <p
                      class="m-0 fw-bold"
                    >
                      Sophia Campbell
                    </p>
                  </div>
                  <div
                    class="col-2"
                  />
                </div>
              </button>
            </div>
          </div>
          <div
            class="modal-footer"
          >
            <button
              class="btn btn-secondary"
              type="button"
            >
              Close
            </button>
            <button
              class="btn btn-primary"
              disabled=""
              type="button"
            >
              Confirm
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
